### PR TITLE
Enabled beaker provisioning for pytest-ansible tests

### PR DESCRIPTION
**Pull request by cpinjani**
Signed-off-by: Chandan Pinjani <cpinjani@redhat.com>

### CODE REVIEW

**Comment by cpinjani** [comment]
If PSI provisioning fails, this enables beaker provisioning by default.

Tested by running pipeline over forked branch:
PSI: https://gitlab.com/dpunia/pki/-/pipelines/312408724
Beaker: https://gitlab.com/dpunia/pki/-/pipelines/312531159

**Comment by dpuniaredhat** [comment]
@cpinjani 1 point, Can we also add the parameter "provision_beaker : true" to run the jobs only on beaker because as observation from last openstack issue sometime machine inaccessible after running few test or packet loss which cause machine hang in between after provisioning on openstack.

We can make condition like.
1. provision_beaker : false it will first try on openstack in-case failure next try out on beaker provisioning. 
2. provision_beaker : true only go for beaker provisioning. 

Rest LGTM

File: tests/dogtag/pytest-ansible/.gitlab-ci.yml [hunk 1]
```diff
@@ -1,4 +1,5 @@
-image: bbhavsar/dogtagpki-test-essentials-container
+#image: bbhavsar/dogtagpki-test-essentials-container
+image: docker-registry.upshift.redhat.com/pki/dogtagpki-test-essentials-container
```
Comment (edewata) [inline tests/dogtag/pytest-ansible/.gitlab-ci.yml:2]: Is this image available publicly?

Comment (cpinjani) [reply tests/dogtag/pytest-ansible/.gitlab-ci.yml:2]: No, its in Red Hat registry

Comment (edewata) [reply tests/dogtag/pytest-ansible/.gitlab-ci.yml:2]: Does the image contain any sensitive information? If not, could we put it https://quay.io/organization/dogtagpki ?

Comment (cpinjani) [reply tests/dogtag/pytest-ansible/.gitlab-ci.yml:2]: Actually yes, its has information pertaining to certs/keytab files for authenticating to Beaker

**Comment by cpinjani** [comment]
> @cpinjani 1 point, Can we also add the parameter "provision_beaker : true" to run the jobs only on beaker because as observation from last openstack issue sometime machine inaccessible after running few test or packet loss which cause machine hang in between after provisioning on openstack.
> 
> We can make condition like.
> 
> 1. provision_beaker : false it will first try on openstack in-case failure next try out on beaker provisioning.
> 2. provision_beaker : true only go for beaker provisioning.
> 
> Rest LGTM

Implemented this. Added 'provision_beaker' as pipeline variable.
If false: it tries first on openstack, in-case of failure next try out on beaker provisioning
If true: Run all jobs using beaker provisioning

provision_beaker - false: https://gitlab.com/dpunia/pki/-/jobs/1314278941
provision_beaker - false, PSI Down: https://gitlab.com/dpunia/pki/-/jobs/1311912037
provision_beaker - true: https://gitlab.com/dpunia/pki/-/jobs/1311560825

File: tests/dogtag/pytest-ansible/.gitlab-ci.yml [hunk 2]
```diff
@@ -30,255 +32,274 @@ before_script:
...
 #------------------------------------------------------------------Tier1 Jobs ---------------------------------------------------------------------------------------------------------
 topo-01-role-user-creation:
   stage: tier1
   <<: *job_definition
   script:
-    - ansible-playbook -u fedora --become -i $HOSTFILE $PYTEST_DIR/installation/main.yml --extra-vars "topology=topology-01" -vv  | sed 's/\\n/\n/g'
-    - py.test --html=$CI_PROJECT_DIR/${CI_JOB_NAME}_html_report.html --self-contained-html --ansible-user=fedora --ansible-become --ansible-inventory $HOSTFILE --ansible-host-pattern master $PYTEST_DIR/pytest/sanity/test_role_users.py -q -s --junitxml $CI_PROJECT_DIR/role-user-creation_junit.xml -vv
+    - ansible-playbook -i $HOSTFILE $PYTEST_DIR/installation/main.yml --extra-vars "topology=topology-01" -vv  | sed 's/\\n/\n/g'
+    - py.test --html=$CI_PROJECT_DIR/${CI_JOB_NAME}_html_report.html --self-contained-html --ansible-inventory $HOSTFILE --ansible-host-pattern master $PYTEST_DIR/pytest/sanity/test_role_users.py -q -s --junitxml $CI_PROJECT_DIR/role-user-creation_junit.xml -vv
+  only:
```
Comment (dpuniaredhat) [inline tests/dogtag/pytest-ansible/.gitlab-ci.yml:72]: @cpinjani remove only before merge

**Review by dpuniaredhat — APPROVED** [review]

**Comment by 06shalini** [comment]
LGTM

**Pull request merged**